### PR TITLE
GUI: Support setting permission on HyperKit machine driver

### PR DESCRIPTION
**Pull request by spowelljr**
Closes https://github.com/kubernetes/minikube/issues/14091

**How It Works:**
If a start fails on Mac, it checks the error message to see if the cause of the error was due to the `docker-machine-driver-hyperkit` not having the proper permissions set. If this is the case we notify the user of the issue via the message box shown below and once they click OK a terminal instance is automatically opened with the command that will apply the required permissions to the file, the user just has to enter their password to execute the command. After they enter the command the terminal closes and the start retries again.

Screen Shot 2022-05-04 at 3 10 06 PM
Screen Shot 2022-05-03 at 11 23 15 AM

### CODE REVIEW

File: gui/window.cpp [hunk 1]
```diff
@@ -931,6 +936,39 @@ void Window::sshConsole()
...
+    process->waitForFinished(-1);
+}
+
+void Window::showHyperKitMessage()
+{
+    QMessageBox dialog;
+    dialog.setWindowTitle(tr("HyperKit Permissions Required"));
+    dialog.setWindowIcon(*trayIconIcon);
+    dialog.setModal(true);
+    dialog.setText(tr(
```
Comment (medyagh) [inline gui/window.cpp:963]: "The HyperKit driver requires a one-time sudo permission, If you like to proceed, Press OK and then enter your password in the terminal the start will resume after."

Comment (medyagh) [reply gui/window.cpp:963]: there should be a cancel button to not continue.

Comment (medyagh) [reply gui/window.cpp:963]: there should be a cancel button to not continue.

**Review by medyagh — CHANGES_REQUESTED** [review]

File: gui/window.cpp (hunk 1 above)
Comment (spowelljr) [reply gui/window.cpp]: Message updated and cancel button added.

[the remark this answers is not in the document]
Comment (spowelljr) [reply gui/window.cpp]: This was included in the previous commit

**Pull request merged**